central/views/imagecveflat/view_impl.go

```diff
@@ -0,0 +1,200 @@
+			return r.CVEIDs[i] < r.CVEIDs[j]
+		})
+		ret = append(ret, r)
+	}
+	return ret, nil
+}
+
+func withSelectCVEIdentifiersQuery(q *v1.Query) *v1.Query {
+	cloned := q.CloneVT()
+	cloned.Selects = []*v1.QuerySelect{
+		search.NewQuerySelect(search.CVEID).Distinct().Proto(),
+	}
+	cloned.GroupBy = &v1.QueryGroupBy{
+		Fields: []string{search.CVE.String()},
+	}
+
+	return cloned
+}
+
+func withSelectCVECoreResponseQuery(q *v1.Query, cveIDsToFilter []string, options views.ReadOptions) *v1.Query {
+	cloned := q.CloneVT()
+	if len(cveIDsToFilter) > 0 {
+		cloned = search.ConjunctionQuery(cloned, search.NewQueryBuilder().AddDocIDs(cveIDsToFilter...).ProtoQuery())
+		cloned.Pagination = q.GetPagination()
+	}
+	cloned.Selects = []*v1.QuerySelect{
+		search.NewQuerySelect(search.CVE).Proto(),
+		search.NewQuerySelect(search.CVEID).Distinct().Proto(),
+	}
+	if !options.SkipGetTopCVSS {
+		cloned.Selects = append(cloned.Selects, search.NewQuerySelect(search.CVSS).AggrFunc(aggregatefunc.Max).Proto())
+	}
+	if !options.SkipGetAffectedImages {
+		cloned.Selects = append(cloned.Selects, search.NewQuerySelect(search.ImageSHA).AggrFunc(aggregatefunc.Count).Distinct().Proto())
+	}
+	if !options.SkipGetFirstDiscoveredInSystem {
+		cloned.Selects = append(cloned.Selects, search.NewQuerySelect(search.CVECreatedTime).AggrFunc(aggregatefunc.Min).Proto())
+	}
+	if !options.SkipPublishedDate {
+		cloned.Selects = append(cloned.Selects, search.NewQuerySelect(search.CVEPublishedOn).AggrFunc(aggregatefunc.Min).Proto())
+	}
+	if !options.SkipGetTopNVDCVSS {
+		cloned.Selects = append(cloned.Selects, search.NewQuerySelect(search.NVDCVSS).AggrFunc(aggregatefunc.Max).Proto())
+	}
+
+	// TODO(ROX-28320): figure out if I want to hide these behind options.
+	cloned.Selects = append(cloned.Selects, search.NewQuerySelect(search.EPSSProbablity).AggrFunc(aggregatefunc.Max).Proto())
+	cloned.Selects = append(cloned.Selects, search.NewQuerySelect(search.ImpactScore).AggrFunc(aggregatefunc.Max).Proto())
+	cloned.Selects = append(cloned.Selects, search.NewQuerySelect(search.FirstImageOccurrenceTimestamp).AggrFunc(aggregatefunc.Min).Proto())
+	cloned.Selects = append(cloned.Selects, search.NewQuerySelect(search.VulnerabilityState).AggrFunc(aggregatefunc.Max).Proto())
+	cloned.Selects = append(cloned.Selects, search.NewQuerySelect(search.Severity).AggrFunc(aggregatefunc.Max).Proto())
+
+	cloned.GroupBy = &v1.QueryGroupBy{
+		Fields: []string{search.CVE.String()},
+	}
+	// TODO(ROX-28320): hack around severity sort for now
+	for _, sortOption := range cloned.GetPagination().GetSortOptions() {
+		if sortOption.Field == search.Severity.String() {
+			log.Infof("SHREWS -- we have Severity sort")
+			sortOption.Field = search.SeverityMax.String()
+		}
+		if sortOption.Field == search.CVSS.String() {
+			log.Infof("SHREWS -- we have CVSS sort")
+			sortOption.Field = search.CVSSMax.String()
+		}
+		if sortOption.Field == search.CVE.String() {
+			log.Infof("SHREWS -- we have CVE sort")
```

Comment:
Nothing to do for this one?

Reply:
yeah only have to do stuff for the aggregation ones.  This one was only here for my debugging.